libs/core/executors/include/hpx/executors/fork_join_executor.hpp

```diff
@@ -208,13 +229,16 @@ namespace hpx { namespace execution { namespace experimental {
                 }
             }
 
-            void wait_state_all(thread_state state)
+            void wait_state_all(thread_state state) const noexcept
             {
                 for (std::size_t t = 0; t < num_threads_; ++t)
                 {
                     while (thread_states_[t].data_.load(
-                               std::memory_order_acquire) != state)
+                               std::memory_order_relaxed) != state)
                     {
+#if defined(HPX_SMT_PAUSE)
+                        HPX_SMT_PAUSE;
+#endif
```

Comment:
Should we maybe always define `HPX_SMT_PAUSE` so that we can remove the `#if defined(...` from here?